code/modules/mob/living/simple_animal/shade.dm

```diff
@@ -124,39 +130,93 @@
+			H.stamina = min(H.max_stamina, H.stamina+(6 * H.stamina_recovery))
+			H.adjustBruteLoss(-1)
+			H.adjustFireLoss(-1)
+			H.adjustToxLoss(-1)
+			H.adjustOxyLoss(-1)
+			adjustCloneLoss(-2)
+			var/list/nagging_doubts = list("You feel empowered by the ancestors!","You feel ancestral might flowing through your veins!","You feel the power of your forebears!", \
+											"You feel the blood of the warrior!", "You feel the glory of a warrior's death!", "You feel mighty!","You feel the strength of the spirits!")
+			if(prob(5) || !possession_heard_message)
+				to_chat(H, "<span class='danger'>[pick(nagging_doubts)]</span>")
+				possession_heard_message = 1
+
+		else
+			if(prob(20) || !possession_heard_message)
+				L.adjustCloneLoss(2)
+				adjustCloneLoss(-2)
+				var/list/nagging_doubts = list("You feel a nagging doubt in the back of your head.","You feel a vacancy in your thoughts.","You feel momentarily forgetful.", \
+												"You feel temporarily occupied.", "You feel a little worried.", "You feel a hostile presence.","You feel watched.")
+				if(prob(5) || !possession_heard_message)
+					to_chat(L, "<span class='notice'>[pick(nagging_doubts)]</span>")
+					possession_heard_message = 1
 
 /mob/living/simple_animal/shade/bluespace/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "", var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
 	..()
-	if(speaker && speaker != src)
+	if(speaker && speaker != src && !possessive)
 		last_message_heard = message
-		message_countdown = min(300, message_countdown + 50)
-		health = min(maxHealth, health + 5)
+		message_countdown = min(300, message_countdown + 35)
+		adjustCloneLoss(-2)
 		if(heard_dying_message)
 			heard_dying_message = 0
 			to_chat(src, "<span class='notice'>The soothing echoes of life reinvigorate you.</span>")
 
 /mob/living/simple_animal/shade/bluespace/say(var/message)
-	var/list/words_in_memory = dd_text2List(last_message_heard, " ")
-	var/list/words_in_message = dd_text2List(message, " ")
-	for(var/word1 in words_in_message)
-		var/valid = 0
-		for(var/word2 in words_in_memory)
-			if(lowertext(word1) != lowertext(word2))
-				continue
-			else
-				valid = 1
-				break
-		if(!valid)
-			message = replacetext(message, word1, pick(words_in_memory))
-	message = slur(message,rand(10, 20))
-	..()
+	if(!possessive)
+		var/new_last_message_heard = sanitizeName(last_message_heard)
+		var/new_message = sanitizeName(message)
+
+		world << new_last_message_heard
+		var/list/words_in_memory = dd_text2List(new_last_message_heard, " ")
+		var/list/words_in_message = dd_text2List(new_message, " ")
+		for(var/word1 in words_in_message)
+			var/valid = 0
+			for(var/word2 in words_in_memory)
+				if(lowertext(word1) != lowertext(word2))
+					continue
+				else
+					valid = 1
+					break
+			if(!valid)
+				message = replacetext(message, word1, pick(words_in_memory))
+		message = slur(message,rand(10, 20))

```

Comment:
This is redundant. Just use ```slur(message,15)```.

Reply:
How so?

Reply:
The second value of proc/slur is already randomized. You're basically calling a rand of a rand with this.

Reply:
```proc/slur(phrase, strength = 100)
	phrase = html_decode(phrase)
	var/leng=lentext(phrase)
	var/counter=lentext(phrase)
	var/newphrase=""
	var/newletter=""
	while(counter>=1)
		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
		if(prob(strength))
			if(rand(1,3)==3)
				if(lowertext(newletter)=="o")	newletter="u"
				if(lowertext(newletter)=="s")	newletter="ch"
				if(lowertext(newletter)=="a")	newletter="ah"
				if(lowertext(newletter)=="c")	newletter="k"
			switch(rand(1,15))
				if(1,3,5,8)	newletter="[lowertext(newletter)]"
				if(2,4,6,15)	newletter="[uppertext(newletter)]"
				if(7)	newletter+="'"
		newphrase+="[newletter]";counter-=1
	return newphrase```

This is proc/slur